picard/script.py

```diff
@@ -299,6 +300,53 @@ def eval(self, script, context=None, file=None):
         return ScriptParser._cache[key].eval(self)
 
 
+class MultiValue(MutableSequence):
+    def __init__(self, parser, multi, separator):
+        assert isinstance(multi, ScriptExpression)
+        self.parser = parser
+        if isinstance(separator, ScriptExpression):
+            self.separator = separator.eval(self.parser)
+        else:
+            self.separator = separator
+        self._multi = self._get_multi_values(multi)
+
+    def _get_multi_values(self, multi):
+        if (self.separator == MULTI_VALUED_JOINER
+            and len(multi) == 1
+            and isinstance(multi[0], ScriptVariable)):
+            # Convert ScriptExpression containing only a single variable into variable
+            return self.parser.context.getall(normalize_tagname(multi[0].name))
+
+        # Fall-back to converting to a string and splitting if haystack is an expression
+        # or user has overridden the separator character.
+        multi = multi.eval(self.parser)
+        if self.separator:
+            return multi.split(self.separator)
+        else:
+            return [multi]
```

Comment:
I found the `return multi.split(separator) if separator else [multi]` one liner in this case even a bit more readable, but no strong preference here.

Reply:
> Only thing that should be added is tests, we should use the chance and introduce some tests for `MultiValue`.

Yes, I'll do that

Reply:
> I found the `return multi.split(separator) if separator else [multi]` one liner in this case even a bit more readable, but no strong preference here.

Since, I used longer variable names. So I prefer to keep it as is.